baseband/asp/base.py

```diff
@@ -0,0 +1,45 @@
+from ..vlbi_base.base import VLBIStreamReaderBase
+from .frame import ASPFrame
+from .header import ASPFileHeader, ASPHeader
+import astropy.units as u
+
+
+__all__ = ['ASPStreamReaderBase', 'ASPStreamReader']
+
+
+class ASPStreamReaderBase(VLBIStreamReaderBase):
+
+    # unfinished
+    def __init__(self, fh_raw, header0):
+        self._fh_raw = fh_raw
+        self._header0 = header0
+        sample_rate = header0['ch_bw'][0] * u.MHz
+        samples_per_frame = header0['NPtsSend'][0]
+        super(ASPStreamReaderBase, self).__init__(
+            fh_raw, header0, sample_rate,
+            samples_per_frame, unsliced_shape=None, bps=8, complex_data=True,
+            squeeze=False, subset=None, fill_value=0.0, verify=False)
+
+    def read_frame(self):
+        frame = ASPFrame.fromfile(self._fh_raw)
+        # associate the original file header with
+        # frame header (just in case!)
+        frame.header.file_header = self._header0.file_header
+        return frame
+
+
+class ASPStreamReader(ASPStreamReaderBase):
+    def __init__(self, fh_raw):
+        pos = fh_raw.tell()
+        fh_raw.seek(0)
+        fileheader0 = ASPFileHeader.fromfile(fh_raw)
+        pos2 = fh_raw.tell()
+        header0 = ASPHeader.fromfile(fh_raw)
+        header0.file_header = fileheader0
+
+        if pos == 0:
+            pos = pos2
+
+        fh_raw.seek(pos)
+
+        super(ASPStreamReader, self).__init__(fh_raw, header0)
```

Comment:
Even as a hack, it would be useful to have an `open` function (not built by `vlbi_base.base.make_opener`, since that allows for 'rb' and 'w' modes).  The reason is because the stream reader currently doesn't open a string filename with `io.open`, so the user has to do it themselves.

`open` can be as simple as this:
```
def open(name):
    # Open file using io if not already opened.
    got_fh = hasattr(name, 'read')
    if not got_fh:
        name = io.open(name, 'rb')
    # Wrap file pointer with stream reader.
    try:
        return ASPStreamReader(name)
    except Exception as exc:
        if not got_fh:
            try:
                name.close()
            except Exception:  # pragma: no cover
                pass
        raise exc
```
This simply tries to open the filename.  If you want automatic file sequence opening, you can do:
```
from ..helpers import sequentialfile as sf

def open(name):
    # If sequentialfile object, check that it's opened properly.
    if isinstance(name, sf.SequentialFileBase):
        assert name.mode == 'rb', (
            "open only accepts sequential files opened in 'rb' "
            "mode for reading.")

    # If passed some kind of list, open a sequentialfile object.
    if isinstance(name, (tuple, list, sf.FileNameSequencer)):
        name = sf.open(name, 'rb')

    # Open file using io if not already opened.
    got_fh = hasattr(name, 'read')
    if not got_fh:
        name = io.open(name, 'rb')
    # Wrap file pointer with stream reader.
    try:
        return ASPStreamReader(name)
    except Exception as exc:
        if not got_fh:
            try:
                name.close()
            except Exception:  # pragma: no cover
                pass
        raise exc
```